fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsMiddleware.kt

```diff
@@ -20,18 +21,26 @@ class MetricsMiddleware(
         next: (AppAction) -> Unit,
         action: AppAction,
     ) {
-        handleAction(action)
+        handleAction(context, action)
         next(action)
     }
 
-    private fun handleAction(action: AppAction) = when (action) {
+    private fun handleAction(context: MiddlewareContext<AppState, AppAction>, action: AppAction) = when (action) {
         is AppAction.AppLifecycleAction.ResumeAction -> {
             metrics.track(Event.GrowthData.SetAsDefault)
             metrics.track(Event.GrowthData.FirstAppOpenForDay)
             metrics.track(Event.GrowthData.FirstWeekSeriesActivity)
             metrics.track(Event.GrowthData.UsageThreshold)
             metrics.track(Event.GrowthData.UserActivated(fromSearch = false))
         }
+        is AppAction.AwesomeBarAction.EngagementFinished -> {
+            for (info in context.state.awesomeBarVisibilityState.suggestionsInfo()) {
+                val impressionInfo = info.suggestion.metadata?.get("impressionInfo") as? FxSuggestImpressionInfo.Amp
+                    ?: continue
+                // TODO: Record Glean impression ping here.
```

Comment:
@tiftran is adding the new `fx-suggest` ping over in #3958. This logic will submit that same ping, but with its `ping_type` set to `fxsuggest-impression` instead of `fxsuggest-click`. We'll also add some new metrics to that ping, for the suggestion position.

So there are three PRs in all (whew!) for Suggest telemetry:

1. #4110, which adds the `AwesomeBar.VisibilityState` instrumentation.
2. #3958, which adds the `fx-suggest` ping (and telemetry for clicks).
3. This PR, which will send an `fx-suggest` ping for impressions.